src/js/components/Drop/DropContainer.js

```diff
@@ -231,19 +288,28 @@ const DropContainer = forwardRef(
       };
 
       let scrollParents;
+      let scrollParentsHorizontal;
```

Comment:
From what I was seeing the scroll horizontal issue was only happening when it was within a containing block that has been transformed. Can we do the scrollParentsHorizontal code only if `dropOptions.checkContainingBlock = true`?

Reply:
(quick/lazy response from me without testing explicitly)

I think even for a case where `getContainingBlock` returns `null` (it has traversed all parents), or the `checkContainingBlock` option is not enabled, we still want the Drop positioning to respond to horizontal scrolling.

For example, say there's a box with `overflow: auto` that has been given an explicit width (that is more than the viewport width) based on the number of children inside of it, and it scrolls horizontally only. Even if we haven't opted in to this new behaviour, we don't get Drop positioning updates when this container is scrolled.

Reply:
I tested with the [code found here](https://codesandbox.io/s/grommet-v2-template-forked-hv4hf6?file=/index.js)m which I think is the situation you are describing. I tested with both a horizontal scroll case and a vertical scroll case. In both situations the drop stays in view when I scroll horizontally or vertically both in the current version of grommet and when testing using the code in this PR. 

Previously in this thread I mentioned that I was seeing the horizontal scroll issue when the drop was within a containing block that had been transformed. I realized that the fix for this was not actually coming from the scrollParentsHorizontal code. When I commented that code out and left the containing block code in I got the same behavior upon scrolling horizontally so the drop is out of view. 

Maybe I am missing the correct test case to see the issue that the scrollParentsHorizontal code is fixing.